File: kraken/src/main/java/glitch/kraken/object/json/Subscriber.java
package glitch.kraken.object.json;

import com.google.gson.annotations.SerializedName;
import glitch.api.objects.enums.SubscriptionType;
import glitch.api.objects.json.interfaces.Creation;
import glitch.api.objects.json.interfaces.IDObject;
import lombok.Data;

import java.time.Instant;

@Data
public class Subscriber implements IDObject<String>, Creation {
    private final String id;
    private final Instant createdAt;
    @SerializedName("sub_plan")
    private final SubscriptionType subscriptionType;
    @SerializedName("sub_plan_name")
    private final String subscriptionName;
    private final User user;
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/Subscriber.kt
package glitch.kraken.`object`.json

import com.google.gson.annotations.SerializedName
import glitch.api.objects.enums.SubscriptionType
import glitch.api.objects.json.interfaces.Creation
import glitch.api.objects.json.interfaces.IDObject
import java.time.Instant

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Subscriber(
        override val id: String,
        override val createdAt: Instant,
        @SerializedName("sub_plan")
        val subscriptionType: SubscriptionType,
        @SerializedName("sub_plan_name")
        val subscriptionName: String,
        val user: User
) : IDObject<String>, Creation
